### PR TITLE
Update libp2p

### PR DESCRIPTION
**Pull request by dimalinux**
Updates our libp2p dependencies to the latest stable versions.

### CODE REVIEW

File: net/discovery.go [hunk 1]
```diff
@@ -10,9 +10,9 @@ import (
 	libp2phost "github.com/libp2p/go-libp2p-core/host"
 	"github.com/libp2p/go-libp2p-core/peer"
 	"github.com/libp2p/go-libp2p-core/peerstore"
-	libp2pdiscovery "github.com/libp2p/go-libp2p-discovery"
```
Comment (dimalinux) [inline net/discovery.go:13]: This is what prevented the upgrade from being trivial. The repo is deprecated and the code was moved into packages of `github.com/libp2p/go-libp2p`:  https://github.com/libp2p/go-libp2p-discovery/commit/111203389c4886581007b0837ba9d48821f4fb6f

File: net/host.go [hunk 2]
```diff
@@ -140,7 +140,7 @@ func NewHost(cfg *Config) (*host, error) {
 	}
 
 	// create libp2p host instance
-	h, err := libp2p.New(context.Background(), opts...)
```
Comment (dimalinux) [inline net/host.go:143]: I guess the context wasn't getting used: https://github.com/libp2p/go-libp2p/commit/b7bee3855cb86e50440e23b463605ea874c38787

File: net/host.go [hunk 3]
```diff
@@ -177,7 +177,6 @@ func (h *host) Start() error {
 	h.h.SetStreamHandler(protocol.ID(h.protocolID+queryID), h.handleQueryStream)
 	h.h.SetStreamHandler(protocol.ID(h.protocolID+swapID), h.handleProtocolStream)
 
-	h.h.Network().SetConnHandler(h.handleConn)
```
Comment (dimalinux) [inline net/host.go:180]: `SetConnHandler` was removed: https://github.com/libp2p/go-libp2p-core/pull/217
We were only using it to log a message, so I just removed it from our code too.

**Review by noot — APPROVED** [review]
looks good! would be nice to eventually add that incoming connection log back in, but I think we will need to implement a libp2p ConnManager for that.

**Pull request merged**